pkg/providers/nutanix/validator.go

```diff
@@ -0,0 +1,195 @@
+	}
+
+	return nil
+}
+
+func (v *Validator) validateImageConfig(ctx context.Context, identifier anywherev1.NutanixResourceIdentifier) error {
+	switch identifier.Type {
+	case anywherev1.NutanixIdentifierName:
+		if identifier.Name == nil || *identifier.Name == "" {
+			return fmt.Errorf("missing image name")
+		} else {
+			imageName := *identifier.Name
+			imageUUID, err := findImageUUIDByName(ctx, v.v3Client, imageName)
+			if err != nil {
+				return fmt.Errorf("failed to find image with name %q: %v", imageName, err)
+			}
+
+			identifier.Type = anywherev1.NutanixIdentifierUUID
+			identifier.UUID = imageUUID
+		}
+	case anywherev1.NutanixIdentifierUUID:
+		if identifier.UUID == nil || *identifier.UUID == "" {
+			return fmt.Errorf("missing image uuid")
+		} else {
+			imageUUID := *identifier.UUID
+			if _, err := v.v3Client.GetImage(ctx, imageUUID); err != nil {
+				return fmt.Errorf("failed to find image with uuid %s: %v", imageUUID, err)
+			}
+		}
+	default:
+		return fmt.Errorf("invalid cluster identifier type: %s; valid types are: %q and %q", identifier.Type, anywherev1.NutanixIdentifierName, anywherev1.NutanixIdentifierUUID)
+	}
+
+	return nil
+}
+
+func (v *Validator) validateSubnetConfig(ctx context.Context, identifier anywherev1.NutanixResourceIdentifier) error {
+	switch identifier.Type {
+	case anywherev1.NutanixIdentifierName:
+		if identifier.Name == nil || *identifier.Name == "" {
+			return fmt.Errorf("missing subnet name")
+		} else {
+			subnetName := *identifier.Name
+			subnetUUID, err := findSubnetUUIDByName(ctx, v.v3Client, subnetName)
+			if err != nil {
+				return fmt.Errorf("failed to find subnet with name %s: %v", subnetName, err)
+			} else {
+				identifier.Type = anywherev1.NutanixIdentifierUUID
+				identifier.UUID = subnetUUID
+			}
+		}
+	case anywherev1.NutanixIdentifierUUID:
+		if identifier.UUID == nil || *identifier.UUID == "" {
+			return fmt.Errorf("missing subnet uuid")
+		} else {
+			subnetUUID := *identifier.UUID
+			if _, err := v.v3Client.GetSubnet(ctx, subnetUUID); err != nil {
+				return fmt.Errorf("failed to find subnet with uuid %s: %v", subnetUUID, err)
+			}
+		}
+	default:
+		return fmt.Errorf("invalid cluster identifier type: %s; valid types are: %q and %q", identifier.Type, anywherev1.NutanixIdentifierName, anywherev1.NutanixIdentifierUUID)
+	}
+
+	return nil
+}
+
+// findSubnetUUIDByName retrieves the subnet uuid by the given subnet name
+func findSubnetUUIDByName(ctx context.Context, v3Client client, subnetName string) (*string, error) {
+	res, err := v3Client.ListSubnet(ctx, &v3.DSMetadata{
+		Filter: utils.StringPtr(fmt.Sprintf("name==%s", subnetName)),
+	})
+	if err != nil || len(res.Entities) == 0 {
+		return nil, fmt.Errorf("failed to find subnet by name %q: %v", subnetName, err)
+	}
+
+	if len(res.Entities) > 1 {
+		logr.FromContextOrDiscard(ctx).Info("Found more than one (%v) subnet with name %q.", len(res.Entities), subnetName)
```

Comment:
Avoid extracting loggers from context. Instead, can we inject it as an explicit parameter so its clear there's a dependency here.

Ditto for elsewhere in the code.